doc/functions/trivial-builders.xml

```diff
@@ -0,0 +1,84 @@
+<section xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xmlns:xi="http://www.w3.org/2001/XInclude"
+         xml:id="sec-trivial-builders">
+ <title>Trivial builders</title>
+
+ <para>
+   There are a couple of functions provide in Nixpkgs that help with
+   building derivations. The most important one,
+   <function>stdenv.mkDerivation</function>, has already been
+   documented above. These wrap
+   <function>stdenv.mkDerivation</function>, making it easier to use
+   in certain cases.
+ </para>
+
+ <variablelist>
+  <varlistentry>
+   <term>
+    <literal>runCommand</literal>
+   </term>
+   <listitem>
+     <para>
+       This takes three arguments, <literal>name</literal>,
+       <literal>env</literal>, and <literal>buildCommand</literal>.
+       <literal>name</literal> is just the name that Nix will use to
+       refer to the derivation. <literal>env</literal> is an attribute
+       set specifying environment variables that will be set for this
+       derivation. <literal>buildCommand</literal> specifies the
+       commands that will be run to create this derivation. Note that
+       you will need to create <literal>$out</literal> for Nix to
+       register the command as successful.
+    </para>
+   </listitem>
+  </varlistentry>
+  <varlistentry>
+   <term>
+    <literal>runCommandCC</literal>
+   </term>
+   <listitem>
+     <para>
+       This works just like <literal>runCommand</literal>. The only
+       difference is that it also provides a C compiler for your use.
+       To minimize your dependencies, you should only use this if you
+       are sure you will need a C compiler as part of running your command.
+    </para>
+   </listitem>
+  </varlistentry>
+  <varlistentry>
+   <term>
+    <literal>writeTextFile</literal>
```

Comment:
maybe mention `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin` and say those are specialization of `makeTextFile`.

I see those have doc and examples attached in `trivial-builders.nix`, maybe it should be possible to use some autodocs here?

Reply:
cc @tazjin

Reply:
I missed this thread - sorry!

@danbst It should be possible to use [nixdoc](https://github.com/tazjin/nixdoc) for this, but the current manual structure only includes doc generation for functions in the `lib` folder. It would be nice to do it for functions in the package set (such as the trivial builders), too. Ping me here or on IRC if you want some pointers!